File: guides/micronaut-security-session-database-authentication/java/src/main/java/example/micronaut/entities/UserRole.java
package example.micronaut.entities;

import io.micronaut.data.annotation.EmbeddedId;
import io.micronaut.data.annotation.MappedEntity;

@MappedEntity // <1>
public class UserRole {

    @EmbeddedId // <2>
    private final UserRoleId id;

    public UserRole(UserRoleId id) {
        this.id = id;
    }

    public UserRoleId getId() {
        return id;
    }
}

File: guides/micronaut-security-session-database-authentication/python/src/example/micronaut/entities/user_role.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import EmbeddedId, MappedEntity
from micronaut.serde.annotation import Serdeable

from .user_role_id import UserRoleId


@dataclass
@Serdeable
@MappedEntity  # <1>
class UserRole:
    userRoleId: Annotated[UserRoleId, EmbeddedId]  # <2>
